src/inference/dev_api/threading/ie_thread_safe_containers.hpp

```diff
@@ -62,7 +62,9 @@ class ThreadSafeBoundedPriorityQueue {
         return false;
     }
     bool try_pop(T& value) {
-        return _pqueue.try_pop(value);
+        if (_capacity)
```

Comment:
Do we also need check _pqueue.empty() ?

Reply:
does not `tbb::concurrent_priority_queue` handle such cases by itself?
Suggest to shorten the statement to:
```c++
return _capacity ? _pqueue.try_pop(value) : false;
```

Reply:
thanks, will simplify the function.
tbb::concurrent_priority_queue itself can handle push, try_pop or emplace thread-safe way.
here we are wrapping the priority queue featured with capacity, which will then be used in auto work loads scheduling.

Reply:
> Do we also need check _pqueue.empty() ?
I suppose no.  see try_pop definition here:
//! Gets a reference to and removes highest priority element
    /** **If a highest priority element was found, sets elem and returns true,
        otherwise returns false.**
        This operation can be safely used concurrently with other push, try_pop or emplace operations. */
    bool try_pop(reference elem) {
        cpq_operation op_data(POP_OP);
        op_data.elem = &elem;
        my_aggregator.execute(&op_data);
        return op_data.status==SUCCEEDED;
    }